src/main/java/com/example/weatherwaveapi/service/YandexWeatherService.java

```diff
@@ -73,9 +77,49 @@ private YandexApiContainer getYandexApiContainer(String url, String format) {
         } catch (HttpClientErrorException | HttpServerErrorException e) {
             log.error(EXCEPTION_MESSAGE, e.getMessage());
             return YandexApiContainer.builder()
+                    .success(false)
                     .errorMessage(format)
                     .build();
         }
     }
 
+    private List<CoordinateWeatherRequest> convertCoordinatesRequest(List<String> coordinates) {
+        if (coordinates == null || coordinates.isEmpty()) {
+            System.err.println("Invalid coordinates list");
+            return Collections.emptyList();
+        }
+        return coordinateBuilder(coordinates);
+    }
+
+    private List<CoordinateWeatherRequest> coordinateBuilder(List<String> coordinates) {
+        List<CoordinateWeatherRequest> result = new ArrayList<>();
+        List<String> errors = new ArrayList<>();
```

Comment:
this is an overkill solution, just let's log errors as they appear

Reply:
You mean just change errors printing to logging?